File: backend/app/models/transaction.py
import pandas as pd
from pydantic import BaseModel

class Transaction(BaseModel):#set name of class to call it 
    date: str
    amount: float
    opponent: str
    opponent_account: str
    comment: str
    own_account: str
    # def __init__(self, date, amount, opponent, opponent_account, comment, own_account):#func set ver
    #     self.date = pd.to_datetime(date, format='%d/%m/%Y')
    #     self.amount = round(amount,2)
    #     self.opponent = opponent
    #     self.opponent_account = opponent_account
    #     self.comment = comment
    #     self.own_account = own_account

    def get_date(self):
        return self.date

    def get_amount(self):
        return self.amount

    def get_positiv_expense(self):
        if(amount >= 0):
            return amount
        return 0

File: dotnet/dotnet/Models/Transaction.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace dotnet.Models
{
    public class Transaction
    {
        [Key]
        public Guid Id { get; set; }
        public DateTime Date { get; set; }

        [Required]
        public Double Amount { get; set; }
        public String OpponentName { get; set; }
        public String OpponentAccount { get; set; }
        public String Comment { get; set; }
        public String OwnAccount { get; set; }
    }
}
